src/components/MapView/index.tsx

```diff
@@ -41,6 +46,11 @@ function MapView({ classes }: MapViewProps) {
     map: { latitude, longitude, zoom },
   } = appConfig;
 
+  useEffect(() => {
+    setpopupData({});
+    setshowPopup(false);
+  }, [layers]);
+
```

Comment:
All this state living here is code smell. This should probably all live in Redux, we don't generally want to combine a bunch of component state with redux state. If this lived in Redux, we wouldn't need an effect listener here - you could just hide the popup by hand when the layer change events get fired.